cypress/e2e/po/components/button-group.po.ts

```diff
@@ -0,0 +1,20 @@
+import ComponentPo from '~/cypress/e2e/po/components/component.po';
+import { CypressChainable } from '~/cypress/e2e/po/po.types';
+
+export default class ButtonGroupPo extends ComponentPo {
+  static byLabel(self: CypressChainable, label: string): ButtonGroupPo {
+    return new ButtonGroupPo(
+      self.find('.btn-group > .btn').contains(label)
+    );
+  }
+
+  // Click on the button
+  set(): Cypress.Chainable {
+    return this.self().click();
+  }
+
+  // Check is button is highlighted
+  isSelected(): Cypress.Chainable {
+    return this.self().parent().should('have.class', 'bg-primary');
```

Comment:
I haven't found a way to verify that the button is highlighted (i.e. contains `bg-primary`) without locating the button using label first and then going to the parent. It has to be done that way otherwise it will grab all of the buttons that are highlighted. I did make the updates you suggested for `set` and `isSelected` to find the btn by label

update made here: https://github.com/rancher/dashboard/pull/8705/commits/f458902608e901f7f1c66493a12d9f5ef13ceba4

Reply:
Sorry, re-reading my comments now and there's a ton of typos. *Reaching out of the instance to a parent to check state feels a bit odd. 

After updates though you've got this right. The button group instance's `self` is the `ButtonGroup` rather than the button.